sagemaker-python-sdk/keras_script_mode_pipe_mode_horovod/tensorflow_keras_CIFAR10.ipynb

```diff
@@ -0,0 +1,618 @@
+   "outputs": [],
+   "source": [
+    "!python generate_tensorboard_command.py"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can install [TensorBoard](https://github.com/tensorflow/tensorboard) locally using `pip install tensorboard`\n",
+    "To access an S3 log dir, you need to configure TensorBoard default region, you can do that by configuaring an environment variable AWS_REGION=region_name, e.g. `export AWS_REGION = 'us-east-1'`\n",
+    "\n",
+    "You can access TensorBoard locally at http://localhost:6006\n",
+    "\n",
+    "Based on the TensorBoard metrics, we can see that:\n",
+    "1. All jobs run for 60 epochs (0 - 59)\n",
+    "2. File mode and Pipe mode run for ~7 minutes - Pipemode doesn't effect training performance\n",
+    "3. Distributed mode run for ~2 minutes\n",
+    "4. All of the jobs trained to a similiar validation accuracy\n",
+    "\n",
+    "This example is runing with a small dataset (179M), for larger datasets, by not copying the data, pipemode can reduce the training time significantly"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Deploy the trained model\n",
+    "\n",
+    "The deploy() method creates an endpoint which serves prediction requests in real-time.\n",
+    "The model saves keras artifacts, Thus, can't use TensorFlow serving.\n",
+    "2 relevant options are available:\n",
+    "1. Export the artifacts to the SavedModel format, [blog post](https://aws.amazon.com/blogs/machine-learning/deploy-trained-keras-or-tensorflow-models-using-amazon-sagemaker/)\n",
+    "2. Use the SKLearn container to load a keras model\n",
+    "\n",
+    "I decided to choose the SKLearn container for this example"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Using to inference.py for inference."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!cat inference_source_dir/inference.py"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sagemaker.sklearn import SKLearnModel\n",
+    "model = SKLearnModel(model_data=estimator.model_data,\n",
+    "                     role=role,\n",
+    "                     source_dir='inference_source_dir',\n",
+    "                     entry_point='inference.py'\n",
+    "                    )"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "predictor = model.deploy(endpoint_name='cifar10',initial_instance_count=1, instance_type='ml.m4.xlarge')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Make some predictions\n",
+    "To verify the endpoint's functionality, we'll start by simply generate random data in the correct shape and make a prediction."
```

Comment:
"To verify the that the endpoint functions properly, we generate random data in the correct shape and get a prediction."